README.md

```diff
@@ -1,34 +1,249 @@
+  - A sample path is './docs'
+
+### baseUrl (MarkBind CLI arguments)
+The base URL relative to your domain.
+- '/reponame'
+  - Defaults to the value of `baseUrl` in your `site.json` file
+  - This is important for deploying your site to GitHub Pages
+  - Note that you will need to specify this or in the site config file (typically the `site.json`), in order to configure the relative URL correctly.
+
+### siteConfig (MarkBind CLI arguments)
+The site config file to use.
+- 'site.json'
+  - This is the default value
+
+# Usage
+In essence, there are two parts to a GitHub Action workflow:
+- The trigger event
+- The jobs/steps to be run after the trigger event occurs
+
+For our context, there two typical trigger events. This is written at the start of the workflow files:
+(Assuming 'master' is the target branch)
+1. Trigger the action whenever there is a push to the repository
+```yaml
+on:
+  push:
+    branches:
+      - master
+```
+
+2. Trigger the action whenever there is a pull request to the repository
+```yaml
+on:
+  pull_request:
+    branches:
+      - master
+```
+
+Then, specify a step to use this action:
+
+E.g.
+```yaml
+name: MarkBind Action
 
-Make sure that your MarkBind project is on the master branch.
+on:
+  push:
+    branches:
+      - master
+
+jobs: 
+  build_and_deploy:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Build & Deploy to GitHub Pages
+        uses: MarkBind/markbind-action@master
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          baseUrl: '/mb-test'
+          version: '3.1.1'
+```
+The above script builds the site from the repository's root directory, with baseUrl of '/mb-test' ('mb-test' is the repository name), with MarkBind version 3.1.1.
 
-1. On the top of your repo, click **Actions**
-2. Click **New workflow**
-3. Click **Set up workflow yourself** on the top right corner
-4. Configure your `main.yml` as follows (note that you need to specify the **BASE_URL** of your GitHub pages)
+Then, it will deploy the site to GitHub Pages. It runs everytime there is a push to the repository.
 
+# Scenarios
+## Deploy to Github Pages on push to main
 ```yaml
-name: MarkBind Deploy
+name: MarkBind Action
+
+on:
+  push:
+    branches:
+      - main
+
+jobs: 
+  build_and_deploy:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Build & Deploy to GitHub Pages
+        uses: MarkBind/markbind-action@master
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          version: '3.1.1'
+```
+
+## Deploy to Surge.sh
+```yaml
+name: MarkBind Action
+
+on: [push]
+
+jobs: 
+  build_and_deploy:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Build & Deploy to GitHub Pages
+        uses: MarkBind/markbind-action@master
+        with:
+          token: ${{ secrets.SURGE_TOKEN }}
+          service: 'surge'
+          domain: 'mb-test.surge.sh'
+```
+Note that if you are using custom domain, you will need to ensure that it is configured with Surge.sh.
```

Comment:
Let's move this note up to the `domain` option. Perhaps we could also add a note on setting up subdomains for pr previews (I'm assuming, not too familiar with surge).